service/src/main/java/bio/terra/profile/db/ProfileChangeLogDao.java

```diff
@@ -0,0 +1,154 @@
+package bio.terra.profile.db;
+
+import bio.terra.common.db.ReadTransaction;
+import bio.terra.common.db.WriteTransaction;
+import bio.terra.common.exception.SerializationException;
+import bio.terra.profile.model.ChangeLogEntry;
+import bio.terra.profile.service.profile.model.BillingProfile;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.time.ZoneOffset;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class ProfileChangeLogDao {
+  private static final Logger logger = LoggerFactory.getLogger(ProfileChangeLogDao.class);
+
+  private final NamedParameterJdbcTemplate jdbcTemplate;
+  private final ObjectMapper objectMapper;
+
+  public static final String CHANGELOG_TABLE = "billing_profile_changelog";
+
+  public static final String ID = "id";
+  public static final String PROFILE_ID = "profile_id";
+  public static final String CHANGES = "changes";
+  public static final String CHANGE_DATE = "change_date";
+  public static final String CHANGE_BY = "change_by";
+  public static final String CHANGE_TYPE = "change_type";
+
+  private static final List<String> COLUMNS =
+      List.of(ID, PROFILE_ID, CHANGE_DATE, CHANGE_TYPE, CHANGE_BY, CHANGES);
+  private static final String SQL_SELECT_LIST = String.join(", ", COLUMNS);
+
+  private static final String INSERT_INTO_TABLE = "INSERT INTO " + CHANGELOG_TABLE;
+  private static final String SQL_GET_BY_PROFILE_ID =
+      "SELECT %s FROM %s WHERE %s = :%s"
+          .formatted(SQL_SELECT_LIST, CHANGELOG_TABLE, PROFILE_ID, PROFILE_ID);
+
+  @Autowired
+  public ProfileChangeLogDao(NamedParameterJdbcTemplate jdbcTemplate, ObjectMapper objectMapper) {
+    this.objectMapper = objectMapper;
+    this.jdbcTemplate = jdbcTemplate;
+  }
+
+  @ReadTransaction
+  public List<ChangeLogEntry> getChangesByProfile(UUID profileId) {
+    var params = new MapSqlParameterSource().addValue(PROFILE_ID, profileId);
+    return jdbcTemplate.query(SQL_GET_BY_PROFILE_ID, params, new ChangeLogMapper(objectMapper));
```

Comment:
This will almost certainly be consumed by user's for rendering into a list (someday). Given that, a default ASC or DESC ordering by CHANGE_DATE would be good.